pkg/dataplane/conntrack.go

```diff
@@ -0,0 +1,114 @@
+package dataplane
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/kube-network-policies/pkg/network"
+)
+
+var (
+	mapIPFamilyToString = map[uint8]v1.IPFamily{
+		unix.AF_INET:  v1.IPv4Protocol,
+		unix.AF_INET6: v1.IPv6Protocol,
+	}
+	mapProtocolToString = map[uint8]v1.Protocol{
+		unix.IPPROTO_TCP:  v1.ProtocolTCP,
+		unix.IPPROTO_UDP:  v1.ProtocolUDP,
+		unix.IPPROTO_SCTP: v1.ProtocolSCTP,
+	}
+)
+
+func PacketFromFlow(flow *netlink.ConntrackFlow) *network.Packet {
+	if flow == nil {
+		return nil
+	}
+	packet := network.Packet{
+		SrcIP:   flow.Forward.SrcIP,
+		DstIP:   flow.Reverse.SrcIP,
+		SrcPort: int(flow.Forward.SrcPort),
+		DstPort: int(flow.Reverse.SrcPort),
+	}
+
+	if family, ok := mapIPFamilyToString[flow.FamilyType]; ok {
+		packet.Family = family
+	} else {
+		klog.V(4).Info("Unknown IP family", "family", flow.FamilyType, "flow", flow)
+		if flow.Forward.SrcIP.To4() != nil {
+			packet.Family = v1.IPv4Protocol
+		} else {
+			packet.Family = v1.IPv6Protocol
+		}
+	}
+
+	if protocol, ok := mapProtocolToString[flow.Forward.Protocol]; ok {
+		packet.Proto = protocol
+	} else {
+		klog.V(4).Info("Unknown protocol", "protocol", flow.Forward.Protocol, "flow", flow)
+		packet.Proto = v1.ProtocolTCP
```

Comment:
This seems possible, but wrong: if `flow.Forward.Protocol` is not `unix.IPPROTO_TCP`, then you shouldn't match it against TCP conntrack entries.

Of course, you don't even call this function unless you've already validated `flow.Forward.Protocol`, so this can't happen anyway. So again, just `packet.Proto = mapProtocolToString[flow.Forward.Protocol]` should be fine.